cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -242,29 +265,66 @@ PredictKernel<algorithmFPType, method, cpu>::~PredictKernel()
 }
 
 template <typename algorithmFPType, prediction::Method method, CpuType cpu>
-services::Status PredictKernel<algorithmFPType, method, cpu>::compute(services::HostAppIface * const pHostApp, const NumericTable * const x,
-                                                                      const decision_forest::classification::Model * const m, NumericTable * const r,
-                                                                      NumericTable * const prob, const size_t nClasses,
-                                                                      const VotingMethod votingMethod)
+Status PredictKernel<algorithmFPType, method, cpu>::compute(services::HostAppIface * const pHostApp, const NumericTable * const x,
+                                                            const decision_forest::classification::Model * const m, NumericTable * const r,
+                                                            NumericTable * const prob, const size_t nClasses, const VotingMethod votingMethod,
+                                                            const HyperparameterType * hyperparameter)
 {
     const daal::algorithms::decision_forest::classification::internal::ModelImpl * const pModel =
         static_cast<const daal::algorithms::decision_forest::classification::internal::ModelImpl * const>(m);
     if (_task == nullptr) _task = new PredictClassificationTask<algorithmFPType, cpu>();
     _task->setParams(x, r, prob, pModel, nClasses, votingMethod);
+
+    if (hyperparameter != nullptr)
+    {
+        DAAL_INT64 blockSizeValue = 0l;
+        Status st                 = hyperparameter->find(blockSize, blockSizeValue);
+        DAAL_CHECK(0l < blockSizeValue, services::ErrorHyperparameterBadValue);
+        if (cpu == avx512)
+        {
+            // TODO: Move this check into decision_forest::detail::infer_ops::check_parameters_ranges.
+            //       Now the information about enabled CPU instructions is not available in infer_ops.
+            DAAL_CHECK(blockSizeValue % (512 / (8 * sizeof(algorithmFPType))) == 0, services::ErrorHyperparameterBadValue);
```

Comment:
Is it possible to give some meaningful names to magic constants?

Reply:
Done. Is it better now?